homework02/allan/string.cc

```diff
@@ -0,0 +1,103 @@
+#include "string.h"
+
+#include <cstring>
+#include <iostream>
+String::String(const char *ch)
+  : String(ch, strlen(ch) + 1)
+{
+}
+
+String::String()
+   :String("", 1)
+{
+}
+
+String::String(const String& other)
+  : String(other.c_str(), other.capacity())
+{
+}
+
+
+String::String(const char *ch, int capacity)
+  : mCapacity(capacity),
+    mData(std::make_unique<char []>(mCapacity))
+{
+  strcpy(mData.get(), ch);
```

Comment:
`std::strcpy`

Reply:
You mean if someone calls with a capacity that is too small? I should have made this ctor private. I didn't really want to expose it. It's basically the implementation of the constructor that the other ones call. That way it'll be safer (safe?)

Reply:
Ignore comment here